src/cpl/share_esmf/IrrigationStreamMod.F90

```diff
@@ -0,0 +1,520 @@
+      SHR_ASSERT_FL( (ubound(irrig_rate_demand_patch,1) >= bounds%endp ), sourcefile, __LINE__)
+      SHR_ASSERT_FL( (lbound(irrig_rate_demand_patch,1) <= bounds%begp ), sourcefile, __LINE__)	
+      SHR_ASSERT_FL( (lbound(irrig_rate_demand_patch,2) == 1 ),                        sourcefile, __LINE__)
+!      SHR_ASSERT_FL( (ubound(irrig_rate_demand_patch,2) >= 64 ),          sourcefile, __LINE__)
+	  
+      ! Get pointer for stream data that is time and spatially interpolate to model time and grid
+
+      ! Place all data from each type into a temporary 2d array
+      lsize = bounds%endg - bounds%begg + 1
+	
+	  ! Read variables based on stream_varnames
+      do n = 1,numFields
+         call dshr_fldbun_getFldPtr(sdat_irrig%pstrm(1)%fldbun_model, trim(stream_varnames(n)), &
+              fldptr2=dataptr2d,  rc=rc)
+         if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+            call ESMF_Finalize(endflag=ESMF_END_ABORT)
+         end if
+
+         ! Read data from data streams
+         select case (stream_varnames(n))
+         case ('sfc_irrig_rate_patch')
+            allocate(irrig_rate_prescribed(bounds%begg:bounds%endg,num_irrig_cft))    
+            do g = bounds%begg, bounds%endg
+               ig = g_to_ig(g)
+               irrig_rate_prescribed(g,:) = dataptr2d(:,ig)
+            enddo
+         case ('irrig_method_patch')
+            allocate(irrig_method_prescribed(bounds%begg:bounds%endg,num_irrig_cft))
+            do g = bounds%begg, bounds%endg
+               ig = g_to_ig(g)
+               irrig_method_prescribed(g,:) = dataptr2d(:,ig)
+            enddo
+         case ('irrig_duration_column')
+            allocate(irrig_duration(bounds%begg:bounds%endg,num_irrig_cft))
+            do g = bounds%begg, bounds%endg
+               ig = g_to_ig(g)
+               irrig_duration(g,:) = INT(dataptr2d(:,ig))
+            enddo
+         case ('irrig_start_time_column')
+            allocate(irrig_start_time(bounds%begg:bounds%endg,num_irrig_cft))
+            do g = bounds%begg, bounds%endg
+               ig = g_to_ig(g)
+               irrig_start_time(g,:) = INT(dataptr2d(:,ig))
+            enddo
+         case ('crop_type')
+            allocate(irrig_crop_type(bounds%begg:bounds%endg,num_irrig_cft))
+	        do g = bounds%begg, bounds%endg
+               ig = g_to_ig(g)
+               irrig_crop_type(g,:) = dataptr2d(:,ig)
+            enddo
+         case ('irrig_longitude')
+            allocate(irrig_lon(bounds%begg:bounds%endg,num_irrig_cft))
+            do g = bounds%begg, bounds%endg
+               ig = g_to_ig(g)
+               irrig_lon(g,:) = dataptr2d(:,ig)
+            enddo
+         case ('irrig_latitude')
+            allocate(irrig_lat(bounds%begg:bounds%endg,num_irrig_cft))
+            do g = bounds%begg, bounds%endg
+               ig = g_to_ig(g)
+               irrig_lat(g,:) = dataptr2d(:,ig)
+            enddo
```

Comment:
- [ ] Why are `irrig_longitude` and `irrig_latitude` read from the stream file? Their read-in variables (`irrig_lon` and `irrig_lat`) don't seem to be used.